Mantis 44504 (reopened): fix position write access

ingoj opened the pull request:
fix permission check for userForm (chekWriteAccess) and replace check for uploadUserPicture by call to this function

ingoj committed:
fix position write access

kergomard commented:
Thank you very much @ingoj for the PR !

It is very much appreciated.

While looking through this I realized one more thing: We can get rid of the checks in `ilObjUserGUI::uploadUserPictureObject()` completely. There are just two calls to this function one from `ilObjUserGUI::saveObject()` and the other from `ilObjUserGUI::editObject()`. In both cases the rights have already been checked. To be sure that this function cannot be called directly from the interface, we need to remove the "Object" from its name. So, I would kindly ask you to:

- [ ] Rename the function `ilObjUserGUI::uploadUserPictureObject()`  to `ilObjUserGUI::uploadUserPicture()` and also adapt its usages.
- [ ] Make the function `ilObjUserGUI::uploadUserPictureObject()` `private`.
- [ ] Completely remove the right-checks from the function.

Am I missing something?

Thanks again and best,
@kergomard

ingoj committed:
Merge branch 'ILIAS-eLearning:release_9' into release_9_usermanagement

ingoj committed:
changes to upload user Picture

rename and protect function, skip unneccessary access checks (initiated by maintainer)

ingoj commented:
changes to uploadUserPictures() are implemented as discussed.

kergomard commented:
Thank you very much @ingoj !

Merged and picked to ILIAS 10 and trunk.

Best,
@kergomard
